src/playdate/system.nim

```diff
@@ -33,9 +36,41 @@ proc getSecondsSinceEpoch* (this: ptr PlaydateSys): tuple[seconds: uint, millise
 type PDCallbackFunction* = proc(): int {.raises: [].}
 var updateCallback: PDCallbackFunction = nil
 
+proc runCatchingTyped*[T](fun: () -> T, fatal: bool = false, messagePrefix: string=""): T =
+    try:
+        return fun()
+    except:
+        let exception = getCurrentException()
+        var message: string = ""
+        try:
+            message = &"{messagePrefix}: {getCurrentExceptionMsg()}\n{exception.getStackTrace()}"
+            # replace line number notation from (90) to :90, which is more common and can be picked up as source link in IDEs
+            message = message.replace('(', ':')
+            message = message.replace(")", "")
+        except:
+            message = getCurrentExceptionMsg() & exception.getStackTrace()
+
+        for line in message.splitLines():
+            # Log the error to the console line-by-line, total stack trace might be too long for single call
+            playdate.system.logToConsole(line)
+
+        if fatal:
+            playdate.system.error("FATAL:" & getCurrentExceptionMsg())
+
+proc runCatchingVoid*(fun: () -> void, fatal: bool = false, messagePrefix: string = "") =
```

Comment:
"fun" might not be a very descriptive name for non-Kotlin devs

Reply:
`body` is the usual name used in Nim